cmd/object-api-errors.go

```diff
@@ -703,5 +703,6 @@ func isErrMethodNotAllowed(err error) bool {
 }
 
 func isErrInvalidRange(err error) bool {
-	return errors.As(err, &errInvalidRange)
+	_, ok := err.(InvalidRange)
```

Comment:
@poornas would it be worth to refactor all `errors.As(err, ...)` using proper type casting in order to prevent similar issues in the future?

Reply:
@snorwin - the regression is not because of incorrect usage of errors.As() but lack of unit tests - which is added in this PR.

Reply:
thanks for the clarification